### PR TITLE
squash error handling into debugging-queries

### PR DESCRIPTION
**Pull request by melrchen**

### CODE REVIEW

File: debugging-queries.mdx [hunk 1]
```diff
@@ -168,3 +177,128 @@ def sum_transaction_amount(txns: User.transaction[Transaction.amount]) -> User.s
...
 ![Logs output example](/img/debugging/logs.png)
+
+## Query Error Categories
+
+In using the debugging techniques above, you may encounter a few different categories of errors
+that have common root causes:
+
+### Request error
+
+Request errors are raised before execution of your resolver code.
```
Comment (sjmignot) [inline debugging-queries.mdx:188]:
```suggestion
Request errors are raised before your resolver code executes.
```

File: debugging-queries.mdx [hunk 2]
```diff
@@ -168,3 +177,128 @@ def sum_transaction_amount(txns: User.transaction[Transaction.amount]) -> User.s
...
+They may occur due to invalid feature names in the
```
Comment (sjmignot) [inline debugging-queries.mdx:189]:
```suggestion
They are often caused by invalid feature names in the
```

File: debugging-queries.mdx [hunk 3]
```diff
@@ -168,3 +177,128 @@ def sum_transaction_amount(txns: User.transaction[Transaction.amount]) -> User.s
...
+input or a request that cannot be satisfied by the resolvers you
```
Comment (sjmignot) [inline debugging-queries.mdx:190]:
```suggestion
input or by requests that cannot be satisfied by the resolvers you
```

File: debugging-queries.mdx [hunk 4]
```diff
@@ -168,3 +177,128 @@ def sum_transaction_amount(txns: User.transaction[Transaction.amount]) -> User.s
...
+have defined.
+
+
+### Field error
+
+Field errors are raised while running a feature resolver
+for a particular field.
```
Comment (sjmignot) [inline debugging-queries.mdx:197]: What is meant by for a particular field, here? Is this like what we have sort of moved to calling feature—like a single feature in a feature set? Maybe something like:

"Feature errors are raised when a specific resolver fails"?

File: debugging-queries.mdx [hunk 5]
```diff
@@ -168,3 +177,128 @@ def sum_transaction_amount(txns: User.transaction[Transaction.amount]) -> User.s
...
+For this type of error, you'll find a `feature` and `resolver`
+attribute in the [error type](#chalkerror).
+
+When a feature resolver crashes, you will receive `null` value
```
Comment (sjmignot) [inline debugging-queries.mdx:201]:
```suggestion
When a resolver crashes, you will receive `null` value
```
Maybe generally just resolver? But IDK haha

File: debugging-queries.mdx [hunk 6]
```diff
@@ -168,3 +177,128 @@ def sum_transaction_amount(txns: User.transaction[Transaction.amount]) -> User.s
...
+in the response.
+To differentiate from a resolver returning a `null` value and
+a failure in the resolver, you need to check the error schema.
+
+### Network error
+
+Network errors are thrown outside your resolvers.
+For example, your request was unauthenticated,
```
Comment (sjmignot) [inline debugging-queries.mdx:209]:
```suggestion
They can be caused by unauthenticated requests,
```

File: debugging-queries.mdx [hunk 7]
```diff
@@ -168,3 +177,128 @@ def sum_transaction_amount(txns: User.transaction[Transaction.amount]) -> User.s
...
+connection failed, or an error occurred within Chalk.
```
Comment (sjmignot) [inline debugging-queries.mdx:210]:
```suggestion
connection failures, or errors occurring within Chalk.
```
Should we remove errors occurring within Chalk haha or find some way to say it more delicately haha :)

**Review by sjmignot — APPROVED** [review]

File: debugging-queries.mdx (hunk 7 above)
Comment (melrchen) [reply debugging-queries.mdx:210]: i will ~ refine ~

**Pull request merged**